envoy/access_log/access_log.h

```diff
@@ -92,7 +94,8 @@ class Instance {
   virtual void log(const Http::RequestHeaderMap* request_headers,
                    const Http::ResponseHeaderMap* response_headers,
                    const Http::ResponseTrailerMap* response_trailers,
-                   const StreamInfo::StreamInfo& stream_info) PURE;
+                   const StreamInfo::StreamInfo& stream_info,
+                   AccessLogType access_log_type = AccessLogType::NotSet) PURE;
```

Comment:
Additionally, we will likely also want this field to be present in the `Filter::evaluate` method, though I'm totally fine with that being a separate PR.

Reply:
I planned to do it in a separate PR where I add a filter as well